src/shared/seccomp-util.h

```diff
@@ -1,14 +1,29 @@
 /* SPDX-License-Identifier: LGPL-2.1-or-later */
 #pragma once
 
-#if HAVE_SECCOMP
-#include <seccomp.h> /* IWYU pragma: export */
-#endif
-
 #include "errno-util.h"
 #include "forward.h"
 
 #if HAVE_SECCOMP
+#include <seccomp.h> /* IWYU pragma: export */
+
+#include "dlfcn-util.h"
+
+extern DLSYM_PROTOTYPE(seccomp_api_get);
+extern DLSYM_PROTOTYPE(seccomp_arch_add);
+extern DLSYM_PROTOTYPE(seccomp_arch_exist);
+extern DLSYM_PROTOTYPE(seccomp_arch_native);
+extern DLSYM_PROTOTYPE(seccomp_arch_remove);
+extern DLSYM_PROTOTYPE(seccomp_attr_set);
+extern DLSYM_PROTOTYPE(seccomp_init);
+extern DLSYM_PROTOTYPE(seccomp_load);
+extern DLSYM_PROTOTYPE(seccomp_release);
+extern DLSYM_PROTOTYPE(seccomp_rule_add_array);
+extern DLSYM_PROTOTYPE(seccomp_rule_add_exact);
+extern DLSYM_PROTOTYPE(seccomp_syscall_resolve_name);
+extern DLSYM_PROTOTYPE(seccomp_syscall_resolve_num_arch);
+
+int dlopen_libseccomp(void);
```

Comment:
This should be defined outside of `HAVE_SECCOMP` and return `-EOPNOTSUP` in that case.